DBMS/Db4/DB4.txt

```diff
@@ -0,0 +1,148 @@
+
+Assignment1 -1
+Write a SELECT command to display name of those members who belong to the category as to which member
+�Keshav Sharma� belongs.
+Note Solve the problem using subquery.
+
+
+SELECT  member_name  FROM members  
+where category in (select category from members WHERE member_name='Keshav Sharma');
+
+Assignment1 -2
+Write a SELECT command to display information on the books
+that have not been returned till date. The information should
+include book issue date, title, member name and due date.
+Note Use Correlated Subquery.
+
+
+
+select m.member_name,bi.issue_date,bi.due_date,t.title_name from members m,book_issue bi,titles t ,books b
+ where m.member_id =bi.member_id and  b.title_id=t.title_id and bi.accession_no=b.accession_no and
+(bi.accession_no,bi.member_id) not in(select accession_no,member_id from book_return);
+
+
+
+Assignment1 -3
+Write a SELECT command to display information on the books
+that have been returned after their due dates. The information
+should include book issue date, title, member name and due
+date.
+Note Use Correlated Subquery.
+
+select  m.member_name,t.title_name,bi.issue_date,bi.due_date
+from members m,titles t,book_issue bi ,books b
+where t.title_id=b.title_id and b.accession_no=bi.accession_no
+and m.member_id = bi.member_id and (bi.member_id,bi.accession_no) in (Select member_id,accession_no from book_return
+where (PERIOD_DIFF(date_format(bi.due_date,'%Y%m%d'),date_format(return_date,'%Y%m%d'))0) and return_date'0000-00-00');
+
+
+Assignment1 -4
+Write a SELECT command to display information of those books
+whose price is equal to the most expensive book purchase so far.
+
+
+
+SELECT  from books
+where price = (SELECT MAX(price) FROM books WHERE status = 'booked');
+
+
+Assignment1 -5
+
+Write a SELECT command to display the second maximum price
+of a book.
+
+SELECT price
+FROM books
+WHERE price = (SELECT MAX(price) FROM books WHERE price  (SELECT MAX(price) FROM books));
```

Comment:
select min(price) from books where price in (select top 2 (price) FROM books order by price desc)